sql/core/src/main/scala/org/apache/spark/sql/execution/joins/HashedRelation.scala

```diff
@@ -71,6 +71,16 @@ private[execution] sealed trait HashedRelation extends KnownSizeEstimation {
    */
   def keyIsUnique: Boolean
 
+  /**
+   * is input: Iterator[InternalRow] empty
+   */
+  def inputEmpty: Boolean
+
+  /**
+   * anyNull key exists in input
```

Comment:
Need more context why this is worthwhile to consider: Perhaps to the effect that it is only used in null aware anti join.

Reply:
will do.

Reply:
done